client/src/utils/Store.js

```diff
@@ -116,14 +116,17 @@ export default class API {
    * Updates a model
    * @param {Object} model model to update
    * @param {String} name label to update model
+   * @param {String} newName label to update model name
    */
-  updateModel = async (name, model) => {
+  updateModel = async (name, newName, model) => {
     try {
       await axios.post(`${baseurl}/update`, {
         params: this.flatten(model),
         model_name: name,
       });
-      await Promise.all([this.setModel(name, model), this.getPlotData()]);
+      await Promise.all([this.setModel(newName, model), this.getPlotData()]);
+      await this.cloneModel(name, newName);
+      await this.deleteModel(name);
     } catch (error) {
       console.error(error);
       // better error handling here, some vue event?
```

Comment:
This is a kind of confusing function, because it doesn't just update a model. Is it actually for updating a model or is it only for the case where a model needs to be updated AND a model needs to be renamed at the same time? Either way, the documentation on the function should indicate what it should be used for and when the `setModel` function should be used instead.

Reply:
Why rename is not considered an "update" to a model? I chose to add renaming in this funtion because rename is technically an update. 
I dont understand the confusion around setmodel and updatemodel. setmodel is a local thing while update is a local *and* a server request. That's why setmodel is called in updatemodel.

Reply:
What if you want to update the model but not rename it?